catalog/async/catalog-async-postingestplugin/src/main/java/org/codice/ddf/catalog/async/processingplugin/ProcessingPostIngestPlugin.java

```diff
@@ -0,0 +1,205 @@
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.Response;
+import ddf.catalog.operation.Update;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.ResourceRequestById;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostIngestPlugin;
+import ddf.catalog.resource.Resource;
+import ddf.catalog.resource.ResourceNotFoundException;
+import ddf.catalog.resource.ResourceNotSupportedException;
+
+/**
+ * The {@code ProcessingPostIngestPlugin} is a {@link PostIngestPlugin} that is responsible for
+ * submitting {@link ProcessRequest}s to the {@link ProcessingFramework}.
+ */
+public class ProcessingPostIngestPlugin implements PostIngestPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProcessingPostIngestPlugin.class);
+
+    private static final String POST_PROCESS_COMPLETE =
+            "catalog-async-processing-plugin:post-process-complete";
+
+    private ProcessingFramework processingFramework;
+
+    private CatalogFramework catalogFramework;
+
+    public ProcessingPostIngestPlugin(CatalogFramework catalogFramework,
+            ProcessingFramework processingFramework) {
+        notNull(catalogFramework, "The catalog framework must not be null");
+        notNull(processingFramework, "The processing framework must not be null");
+
+        this.catalogFramework = catalogFramework;
+        this.processingFramework = processingFramework;
+    }
+
+    @Override
+    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+        if (input != null && input.getCreatedMetacards() != null
+                && !isAlreadyPostProcessed(input)) {
+            processingFramework.submitCreate(createCreateRequest(input));
+        }
+        return input;
+    }
+
+    @Override
+    public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+        if (input != null && input.getUpdatedMetacards() != null
+                && !isAlreadyPostProcessed(input)) {
+            processingFramework.submitUpdate(createUpdateRequest(input));
+        }
+        return input;
+    }
+
+    @Override
+    public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
+        if (input != null && input.getDeletedMetacards() != null
+                && !isAlreadyPostProcessed(input)) {
+            processingFramework.submitDelete(createDeleteRequest(input));
+        }
+        return input;
+    }
+
+    private static boolean isAlreadyPostProcessed(Response response) {
+        Map<String, Serializable> properties = response.getRequest()
+                .getProperties();
+        if (properties.containsKey(POST_PROCESS_COMPLETE)) {
+            Serializable prop = properties.get(POST_PROCESS_COMPLETE);
+            if (prop instanceof Boolean) {
+                return (boolean) prop;
+            } else {
+                LOGGER.debug(
```

Comment:
I can up the log level to `Warn`. Now that you mention though, we need to return `true` here and skip `PostProcessPlugin` processing altogether because that would mean something else is setting this property improperly, then when the `ProcessingFramework` sends its results back to the `Catalog` it will hit these plugins again and likely hit this statement again getting stuck in a loop of `PostProcessPlugin` processing.